### PR TITLE
Feat/(#265):- Back to top component.

### PR DESCRIPTION
**Pull request by anjneeksharma**
# Closes/Fixes/Resolves

### 

# Explain the feature/fix

Back to top component it can be used in any application user can add revealAt attribute that describe when user wants to show this back to top, Its main functionality once user click on this it will scroll window to top. 
## Does this PR introduce a breaking change

No 

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [x] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [ ] Does your component follow One Platform style guidelines?
- [ ] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [ ] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [x] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [ ] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

File: packages/opc-back-to-top/.npmignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_temp
```
Comment (hybridx) [inline packages/opc-back-to-top/.npmignore:1]: demo
src
test

File: packages/opc-back-to-top/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# This is opc-back-to-top 👋
+
+A standardized web component based on Lit element for Red Hat One Platform back to top
```
Comment (hybridx) [inline packages/opc-back-to-top/README.md:3]: @anjneeksharma Can you use the new README structure?

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+:host {
+  // add root styles here
+}
+
+.hide{
+  display: none;
+}
+.show {
+  
+  font-family: 'Red Hat Text', sans-serif;
```
Comment (hybridx) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:18]:
```suggestion
.show {
  font-family: 'Red Hat Text', sans-serif;
```

File: packages/opc-back-to-top/src/opc-back-to-top.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  
+}
```
Comment (hybridx) [inline packages/opc-back-to-top/src/opc-back-to-top.scss:23]:
```suggestion
  align-items: center;
}

```

File: packages/opc-back-to-top/test/opc-back-to-top.test.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    it('is defined', async () => {
+        expect(buttonElement).toBeDefined();
+      });
+
+      it('has revealAt defined', async () => {
+        buttonElement.setAttribute('revealAt', '200');
+        await buttonElement.updateComplete;
+        expect(buttonElement.revealAt).toEqual(200);
+      });
+});
```
Comment (hybridx) [inline packages/opc-back-to-top/test/opc-back-to-top.test.ts:32]:
```suggestion
});

```

File: packages/opc-back-to-top/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        }),
+    ],
+    resolve: {
+        extensions: [ '.tsx', '.ts', '.js' ],
+    },
+    output: {
+        filename: 'opc-back-to-top.js',
+        path: path.resolve(__dirname, 'dist'),
+    },
+};
```
Comment (hybridx) [inline packages/opc-back-to-top/webpack.config.js:48]:
```suggestion
};

```

**Review by hybridx — APPROVED** [review]

File: packages/opc-back-to-top/demo/index.html [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    <style> 
+    div{
+        height: 100vh;
+        background-color: #C0C0C0;
+    }
+     </style>
+</head>
+<body>
+    <div></div>
+    <opc-back-to-top revealAt="200"></opc-back-to-top>
```
Comment (hybridx) [inline packages/opc-back-to-top/demo/index.html:17]:
```suggestion
    <opc-back-to-top reveal-at="200"></opc-back-to-top>
```

**Review by hybridx — CHANGES_REQUESTED** [review]

**Review by gisumit — CHANGES_REQUESTED** [review]

File: packages/opc-back-to-top/README.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+## Install
+
+```sh
+npm install
+```
+
+## Usage
+### Back-to-top with revealAt attribute:
+<opc-back-to-top revealAt="200"></opc-back-to-top>
```
Comment (gisumit) [inline packages/opc-back-to-top/README.md:13]: Can you document what this does.

File: packages/opc-back-to-top/README.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+### Back-to-top without revealAt attribute:
+<opc-back-to-top></opc-back-to-top>
```
Comment (gisumit) [inline packages/opc-back-to-top/README.md:16]: Can you add more information... please check the readme of the header and footer component to take reference.

Comment (gisumit) [reply packages/opc-back-to-top/README.md:16]: documentation is missing on the usage in angular and react.

File: packages/opc-back-to-top/package.json [hunk 10]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@one-platform/opc-back-to-top",
+  "version": "0.0.1",
```
Comment (gisumit) [inline packages/opc-back-to-top/package.json:3]: can you add the prerelease tag. to version.

**Comment by anjneeksharma** [comment]
Closing because created on upstream.

**Pull request closed without merging**